browser/ui/views/tabs/brave_tab_strip.cc

```diff
@@ -101,6 +103,26 @@ void BraveTabStrip::MaybeStartDrag(
     }
   }
 
+  if (base::FeatureList::IsEnabled(tabs::features::kBraveSharedPinnedTabs)) {
+    // When source tab is bound for dummy web contents for a shared pinned tab,
+    // we shouldn't kick off drag-and-drop session as the web contents will be
+    // replaced soon.
+    if (source->GetTabSlotViewType() == TabSlotView::ViewType::kTab &&
+        static_cast<Tab*>(source)->data().pinned) {
+      auto index = GetModelIndexOf(source).value();
+      auto* browser = controller_->GetBrowser();
+      DCHECK(browser);
+
+      auto* shared_pinned_tab_service =
+          SharedPinnedTabServiceFactory::GetForProfile(browser->profile());
+      DCHECK(shared_pinned_tab_service);
+      if (shared_pinned_tab_service->IsDummyContents(
```

Comment:
We only block DummyContents because users should still be able to reorder pinned tabs, right? ~~Do we stop them from trying to drag pinned tabs between windows?~~ - I see we do :smile: